Subject: theming-engine: Adjust register_property() to recent glib change

GParamSpec now uses an intern'ed string for 'name', so the code
in gtk_theming_engine_register_property() which prefixes the name
with a namespace has to be adjusted.

https://bugzilla.gnome.org/show_bug.cgi?id=654695
---
diff --git a/gtk/gtkthemingengine.c b/gtk/gtkthemingengine.c
index 66adad6..64a2bc5 100644
--- a/gtk/gtkthemingengine.c
+++ b/gtk/gtkthemingengine.c
@@ -391,8 +391,8 @@ gtk_theming_engine_register_property (const gchar            *name_space,
 
   /* FIXME: hack hack hack, replacing pspec->name to include namespace */
   name = g_strdup_printf ("-%s-%s", name_space, pspec->name);
-  g_free (pspec->name);
-  pspec->name = name;
+  pspec->name = (char *)g_intern_string (name);
+  g_free (name);
 
   gtk_style_properties_register_property (parse_func, pspec);
 }

